sale_order_product_recommendation/models/res_company.py

```diff
@@ -0,0 +1,11 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    force_zero_units_included = fields.Boolean()
+    sale_line_recommendation_domain = fields.Char(default="[]")
```

Comment:
This will lead the user to set an invalid domain.
When creating/writing this value, I would test it with `is_leaf` https://github.com/odoo/odoo/blob/17.0/odoo/osv/expression.py#L712

Reply:
Seems a good idea!  💡 Do you know any example of code with that in OCA?

Reply:
```
from odoo.osv.expression import FALSE_LEAF, OR, is_leaf

if is_leaf(sale_line_recommendation_domain)
```

Reply:
Fortunately, the widget has built-in validation
![image](https://github.com/OCA/sale-workflow/assets/67733397/f4a17e2c-26d3-49dc-9a68-bb451348e513)

Reply:
Checked it with Telmo and the widget for domain editing already incorporates a friendly "invalid domain" message